keras_nlp/layers/modeling/transformer_decoder.py

```diff
@@ -503,3 +503,48 @@ def get_config(self):
 
     def compute_output_shape(self, decoder_sequence_shape):
         return decoder_sequence_shape
+
+    def compute_output_spec(
```

Comment:
Yes, compute_output shape is probably not required when compute_output_spec is implemented. Here compute_output_spec was necessary because the layer returns differently shaped outputs depending on inputs.